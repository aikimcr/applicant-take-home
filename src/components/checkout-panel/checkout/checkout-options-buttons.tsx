import React from 'react';
import { useAppSelector } from '../../../hooks';
import { getCurrentGiftCardIndex, PrizeoutOfferValueOptions } from '../../../slices/offers-slice';
import CheckoutOptionButton from './checkout-option-button';

import './checkout.less';

interface CheckoutOptionsButtonsProps {
    giftcardList: PrizeoutOfferValueOptions[];
    onNewGiftCard: (newIndex: number) => void;
}

export const CheckoutOptionsButtons: React.FC<CheckoutOptionsButtonsProps> = ({
    giftcardList,
    onNewGiftCard,
}): React.ReactElement => {
    const giftCardIndex = useAppSelector(getCurrentGiftCardIndex);

    const returnOptions = () => {
        return giftcardList.map((giftcard, index) => (
            <CheckoutOptionButton
                key={`checkout-options-${giftcard.checkout_value_id}`}
                giftcard={giftcard}
                onClick={() => onNewGiftCard(index)}
                isChecked={index === giftCardIndex}
            />
        ));
    };

    return (
        <section className="checkout-options-buttons">
            <div className="grid">{returnOptions()}</div>
        </section>
    );
};

export default CheckoutOptionsButtons;
