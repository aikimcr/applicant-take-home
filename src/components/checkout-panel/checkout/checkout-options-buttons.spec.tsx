import { render, screen } from '../../../testing/test-utils';
import React from 'react';
import '@testing-library/jest-dom';

import { CheckoutOptionsButtons } from './checkout-options-buttons';
import { PrizeoutOfferValueOptions } from '../../../slices/offers-slice';

describe('Test CheckoutOptionsButtonse', () => {
    const testCards: PrizeoutOfferValueOptions[] = [
        {
            checkout_value_id: 'xyzzy1',
            cost_in_cents: 340,
            display_bonus: 10,
            display_monetary_bonus: null,
            value_in_cents: 374,
        },
        {
            checkout_value_id: 'xyzzy2',
            cost_in_cents: 1020,
            display_bonus: null,
            display_monetary_bonus: 200,
            value_in_cents: 1230,
        },
    ];

    test('Component matches snapshot', () => {
        const newCardHandler = jest.fn();
        const { asFragment } = render(
            <CheckoutOptionsButtons giftcardList={testCards} onNewGiftCard={newCardHandler} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('Change giftcard index', () => {
        const newCardHandler = jest.fn();
        render(<CheckoutOptionsButtons giftcardList={testCards} onNewGiftCard={newCardHandler} />);

        const firstButton = screen.getByText('$3.40');
        firstButton.click();

        expect(newCardHandler).toBeCalledTimes(1);
        expect(newCardHandler).toBeCalledWith(0);
    });
});
