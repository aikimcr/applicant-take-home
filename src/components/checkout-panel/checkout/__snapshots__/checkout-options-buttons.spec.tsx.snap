// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckoutOptionsButtonse Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="checkout-options-buttons"
  >
    <div
      class="grid"
    >
      <div
        class="grid__item"
      >
        <div
          class="checkout-radio"
        >
          <input
            checked=""
            id="checkout-option-button-xyzzy1"
            name="giftcard-option"
            type="radio"
            value="340"
          />
          <label
            for="checkout-option-button-xyzzy1"
          >
            $3.40
          </label>
        </div>
      </div>
      <div
        class="grid__item"
      >
        <div
          class="checkout-radio"
        >
          <input
            id="checkout-option-button-xyzzy2"
            name="giftcard-option"
            type="radio"
            value="1020"
          />
          <label
            for="checkout-option-button-xyzzy2"
          >
            $10.20
          </label>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
